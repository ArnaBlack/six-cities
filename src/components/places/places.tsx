import * as React from 'react';
import {connect} from 'react-redux';

import withActiveItem from '../../hocs/with-active-item/with-active-item';
import withToggleSorting from '../../hocs/with-toggle-sorting/with-toggle-sorting';

import PlaceSorting from '../place-sorting/place-sorting';
import PlaceList from '../place-list/place-list';

import {
  getOffersByCity,
  getCurrentCity,
} from '../../store/data/selectors';
import {SortingTypes} from '../../constants';
import {City, Offer} from "../../types";

const PlaceSortingWrapped = withActiveItem(withToggleSorting(PlaceSorting));

interface Props {
  offers: Offer[],
  currentCity: City,
  onSelectOffer: (offer: Offer | null) => void,
  onSortingTypeChange: (type: string) => void,
}

const Places = (props: Props) => {
  const {
    offers,
    currentCity,
    onSelectOffer,
    onSortingTypeChange,
  } = props;

  return <section className="cities__places places">
    <h2 className="visually-hidden">Places</h2>
    <b className="places__found">{offers.length} places to stay in {currentCity.name}</b>
    <PlaceSortingWrapped
      onChangeType={onSortingTypeChange}
    />
    <PlaceList
      offers={offers}
      onSelectOffer={onSelectOffer}
      listClass="cities__places-list tabs__content"
      cardClass="cities__place-card"
      imageWrapperClass="cities__image-wrapper"
    />
  </section>;
};

const mapStateToProps = (state, props) => ({
  ...props,
  currentCity: getCurrentCity(state),
});

export {Places};
export default connect(mapStateToProps)(Places);
