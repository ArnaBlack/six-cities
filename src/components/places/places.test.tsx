import * as React from 'react';
import {
  configure,
  shallow,
} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import {Places} from './places';

configure({adapter: new Adapter()});

const mock = {
  offers: [],
  currentCity: {
    name: `Brussels`,
    location: {
      latitude: 50.846557,
      longitude: 4.351697,
      zoom: 13,
    },
  },
};

it(`Places correctly renders`, () => {
  const {
    offers,
    currentCity,
  } = mock;
  const onSelectOffer = jest.fn();
  const onSortingTypeChange = jest.fn();
  const tree = shallow(<Places
    offers={offers}
    currentCity={currentCity}
    onSelectOffer={onSelectOffer}
    onSortingTypeChange={onSortingTypeChange}
  />);

  expect(tree).toMatchSnapshot();
});
