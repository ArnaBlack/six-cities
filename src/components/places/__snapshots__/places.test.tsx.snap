// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Places correctly renders 1`] = `
<section
  className="cities__places places"
>
  <h2
    className="visually-hidden"
  >
    Places
  </h2>
  <b
    className="places__found"
  >
    0
     places to stay in 
    Brussels
  </b>
  <WithActiveItem
    onChangeType={[MockFunction]}
  />
  <PlaceList
    cardClass="cities__place-card"
    imageWrapperClass="cities__image-wrapper"
    listClass="cities__places-list tabs__content"
    offers={Array []}
    onSelectOffer={[MockFunction]}
  />
</section>
`;
